src/test/java/sele3/TestBase.java

```diff
@@ -1,18 +1,29 @@
 package sele3;
 
+import com.codeborne.selenide.Configuration;
+import com.sele3.utils.Constants;
+import com.sele3.utils.YamlUtils;
 import lombok.extern.slf4j.Slf4j;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Optional;
 import org.testng.annotations.Parameters;
 
+import java.util.Objects;
+
 import static com.codeborne.selenide.Selenide.closeWebDriver;
 
 @Slf4j
 public class TestBase {
 
     @BeforeClass(alwaysRun = true)
-    public void beforeClass() {
+    @Parameters({"browser", "environment", "language"})
+    public void beforeClass(String browser, String env, @Optional String language) {
+        Configuration.browser = Objects.requireNonNullElse(browser, "chrome");
+        System.setProperty("profile", env);
```

Comment:
you're doing wrong thing. System property will overwrite values from TestNG parameter. So instead of assign value from TestNG param to System property, simply use System property